gems/aws-sdk-core/lib/seahorse/client/net_http/handler.rb

```diff
@@ -74,7 +74,9 @@ def error_message(req, error)
         # @return [void]
         def transmit(config, req, resp)
           session(config, req) do |http|
+            Thread.current[:net_http_skip_default_content_type] = true
             http.request(build_net_request(req)) do |net_resp|
+              Thread.current[:net_http_skip_default_content_type] = nil
```

Comment:
I think this unset should also be done in an `ensure` so that its still unset if there are errors.